js/runner.go

```diff
@@ -368,50 +368,33 @@ type VU struct {
 
 	setupData goja.Value
 
-	// A VU will track the last context it was called with for cancellation.
-	// Note that interruptTrackedCtx is the context that is currently being tracked, while
-	// interruptCancel cancels an unrelated context that terminates the tracking goroutine
-	// without triggering an interrupt (for if the context changes).
-	// There are cleaner ways of handling the interruption problem, but this is a hot path that
-	// needs to be called thousands of times per second, which rules out anything that spawns a
-	// goroutine per call.
-	interruptTrackedCtx context.Context
-	interruptCancel     context.CancelFunc
-
-	m *sync.Mutex
+	activeCtx context.Context
 }
 
 // Verify that VU implements lib.VU
-var _ lib.VU = &VU{}
+var _ lib.InitializedVU = &VU{}
+var _ lib.ActiveVU = &VU{}
```

Comment:
Why do you insist on using the same type for both `InitializedVU` and `ActiveVU`? They are separate interfaces for a reason...

Reply:
What can be wrong with using the same type for both interface? I don't insist, just want to experiment it.